### PR TITLE
Convert to redux, react-router, react-router-redux, and ES6ify everything

### PR DESCRIPTION
**Pull request by browniefed**
Everything is now ES6ified, stateless components, redux, react-redux-rotuer, etc, etc, etc.

### CODE REVIEW

File: js/actions/Plugins.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export const LOAD_PLUGINS = "LOAD_PLUGINS";
+export const LOAD_PLUGINS_SUCCESS = "LOAD_PLUGINS_SUCCESS";
+
+
+export const loadPlugins = () => {
+    return (dispatch) => {
+        dispatch(receiveLoadPlugins());
+        
+        fetch('https://raw.githubusercontent.com/himynameisdave/postcss-plugins/master/plugins.json')
```
Comment (mxstbr) [inline js/actions/Plugins.js:11]: Do we need a polyfill for that or do we have it already?

File: js/components/ListHeading.react.js [hunk 2]
```diff
@@ -1,17 +1,15 @@
-var ListHeading = React.createClass({
-  render: function() {
+import React from "react";
+import { Link } from "react-router";
+
+const ListHeading = ({selectTag, text}) => {
```
Comment (mxstbr) [inline js/components/ListHeading.react.js]: Is the selectTag still necessary here? I can't see another reference in the code to it.

**Comment by mxstbr** [comment]
Holy guacamoly this is awesome, thanks so much Jason! :fire: Two small question and then I'll merge and deploy it. Thanks!!

File: js/actions/Plugins.js (hunk 1 above)
Comment (browniefed) [inline js/actions/Plugins.js:11]: Oops yeah accidentally removed it in the `AppStore`, will add it back in.

File: js/components/ListHeading.react.js (hunk 2 above)
Comment (browniefed) [inline js/components/ListHeading.react.js]: Oh nope, good catch, will remove it.

**Comment by browniefed** [comment]
@mxstbr alright, should to go now.

**Comment by himynameisdave** [comment]
This is actually a convenient way for me to get a handle of integrating `react-redux` and `react-redux-router`, so yeah thanks a ton!

**Comment by browniefed** [comment]
Ah sorry about the conflicts, was working off my outdated master vs upstream merged changes.
Should be fixed now.

**Comment by browniefed** [comment]
@himynameisdave nothing too crazy here, with `react-router-redux` you can just use `Link` as you normally would. However in our case for typing a search term we pull in `push` and `replace` which our action creators so we can live update the URL while someone types.

This also drives the input value, so there is no internal state, the value of the input is controlled entirely by the the URL

**Comment by mxstbr** [comment]
Awesome, thanks so much Jason, will deploy asap!

**Pull request merged**